Library/Homebrew/cmd/linkapps.rb

```diff
@@ -1,12 +1,11 @@
-# Links any Applications (.app) found in installed prefixes to /Applications
 require "keg"
 require "formula"
 
 module Homebrew
   def linkapps
-    target_dir = ARGV.include?("--local") ? File.expand_path("~/Applications") : "/Applications"
+    target_dir = linkapps_target(:local => ARGV.include?("--local"))
 
-    unless File.exist? target_dir
+    unless target_dir.directory?
```

Comment:
Personally I dislike `unless`/`else` as I find it hard to parse. Feel free to ignore me though: not your code.